File: read_docx.js
const fs = require('fs');
const unzipper = require('unzipper');

async function extractTextFromDocx(docxPath) {
    try {
        const directory = await unzipper.Open.file(docxPath);
        const documentXml = directory.files.find(d => d.path === 'word/document.xml');
        if (!documentXml) {
            console.error('word/document.xml not found in the docx file.');
            return;
        }
        const content = await documentXml.buffer();
        const xmlString = content.toString('utf8');
        const text = xmlString.replace(/<[^>]+>/g, ' ').replace(/\s+/g, ' ').trim();
        console.log(text);
    } catch (e) {
        console.error('Error:', e);
    }
}

if (process.argv.length > 2) {
    extractTextFromDocx(process.argv[2]);
}

File: read_docx.py
import zipfile
import xml.etree.ElementTree as ET
import sys
import re

def extract_text_from_docx(docx_path):
    try:
        with zipfile.ZipFile(docx_path, 'r') as docx:
            xml_content = docx.read('word/document.xml')
            
            # Simple regex to remove xml tags
            text = re.sub(rb'<[^>]+>', b' ', xml_content)
            text = re.sub(rb'\s+', b' ', text)
            print(text.decode('utf-8'))
    except Exception as e:
        print("Error:", e)

if __name__ == "__main__":
    if len(sys.argv) > 1:
        extract_text_from_docx(sys.argv[1])
